internal/pkg/agent/vault/vault_keychain_notdarwin.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !darwin
+
+package vault
+
+import (
+	"context"
+	"errors"
+)
+
+var ErrorImplementationNotAvailable = errors.New("vault keychain implementation is not available")
```

Comment:
[Consistency]
We use `Err` instead of `Error` throughout the code, so I'd ask you to keep it consistent.

Reply:
Addressed in d3af9f9d69b4be829de6bb302445d2a95127106b